Components/Queues/Drivers/PhpAmqpLibDriver.php

```diff
@@ -0,0 +1,360 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Smartbox\Integration\FrameworkBundle\Components\Queues\Drivers;
+
+use PhpAmqpLib\Channel\AMQPChannel;
+use PhpAmqpLib\Connection\AbstractConnection;
+use PhpAmqpLib\Connection\AMQPStreamConnection;
+use PhpAmqpLib\Exception\AMQPIOException;
+use PhpAmqpLib\Exception\AMQPProtocolException;
+use PhpAmqpLib\Message\AMQPMessage;
+use PhpAmqpLib\Wire\AMQPTable;
+use Smartbox\Integration\FrameworkBundle\Components\Queues\QueueMessage;
+use Smartbox\Integration\FrameworkBundle\Components\Queues\QueueMessageInterface;
+use Smartbox\Integration\FrameworkBundle\Core\Messages\Context;
+use Smartbox\Integration\FrameworkBundle\DependencyInjection\Traits\UsesSerializer;
+use Smartbox\Integration\FrameworkBundle\Service;
+
+/**
+ * Class PhpAmqpLibDriver.
+ */
+class PhpAmqpLibDriver extends Service implements AsyncQueueDriverInterface
+{
+    use UsesSerializer;
+
+    /**
+     * This field specifies the prefetch window size in octets.
+     * The server will send a message in advance if it is equal to or smaller in size than the available prefetch size
+     * (and also falls into other prefetch limits).
+     * May be set to zero, meaning "no specific limit", although other prefetch limits may still apply.
+     * The prefetch-size is ignored if the no-ack option is set.
+     * This parameter is not implemented in RabbitMQ as checked in:
+     * https://www.rabbitmq.com/specification.html#method-status-basic.qos.
+     */
+    const PREFETCH_SIZE = null;
+
+    /**
+     * Represents the amount of message to consume by iteration.
+     */
+    const PREFETCH_COUNT = 10;
+
+    /*
+     * Default exchange name
+     */
+    const EXCHANGE_NAME = '';
+
+    /**
+     * Default port to connection.
+     */
+    const DEFAULT_PORT = 5672;
+
+    /**
+     * Default host to connect.
+     */
+    const DEFAULT_HOST = 'localhost';
+
+    /**
+     * Default value to connection timeout
+     */
+    const CONNECTION_TIMEOUT = 3.0;
+
+    /**
+     * Default value to read timeout
+     */
+    const READ_TIMEOUT = 10;
```

Comment:
Based on what you discovered of the READ_TIMEOUT is recommended to be  I would add the default value to 130 rather than 10

Reply:
Based on this changes here https://github.com/php-amqplib/php-amqplib/pull/648/files/84556d2027316c49825ee2fa01ea4907f7ab29e1. As you see in this comment, the read_timeout must be greater than 2x the heartbeat. So, if heartbeat value will be defined to 90 as default, we should set a value like +-200.